security/user_provider.rst

```diff
@@ -215,6 +215,11 @@ It's not recommended to use this provider in real applications because of its
 limitations and how difficult it is to manage users. It may be useful in application
 prototypes and for limited applications that don't store users in databases.
 
+.. caution::
+
+    When using a ``memory`` provider, the :class:`Symfony\\Component\\Security\\Core\\User\\User`
```

Comment:
`... the User and ...`  reads a bit weird -> `... the User class and ...`  ?

Reply:
@javiereguiluz You're right, see #13196. I thought I had written it this way (at least in my head). Thanks